app/models/template_invocation.rb

```diff
@@ -1,13 +1,24 @@
 class TemplateInvocation < ActiveRecord::Base
+  include ForemanTasks::Concerns::ActionSubject
 
   belongs_to :template, :class_name => 'JobTemplate', :foreign_key => 'template_id'
   belongs_to :job_invocation, :inverse_of => :template_invocations
   has_many :input_values, :class_name => 'TemplateInvocationInputValue', :dependent => :destroy
   has_one :targeting, :through => :job_invocation
 
-  include ForemanTasks::Concerns::ActionSubject
+  validates_associated :input_values
+  validate :provides_required_input_values
 
   def to_action_input
     { :id => id, :name => template.name }
   end
+
+  private
+
+  def provides_required_input_values
+    required_input_ids = self.template.template_inputs.where(:required => true).pluck(:id)
+    unless (required_input_ids - self.input_values.pluck(:template_input_id)).empty?
```

Comment:
this does not work with in-memory objects that are not persisted yet, when I do a new job_invocation and specify values, template invocations are invalid since no input value is found

if you do `.map(&:template_input_id)` it works